src/Controller/OPDS/OpdsController.php

```diff
@@ -0,0 +1,182 @@
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Attribute\Route;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+#[Route('/opds/{accessKey}', name: 'opds_')]
+class OpdsController extends AbstractController
+{
+    public function __construct(private readonly Opds $opds, private readonly BookRepository $bookRepository, private readonly SpecificCollectionFinder $bookFinder)
+    {
+    }
+
+    private function generateOpdsUrl(string $route, array $params = []): string
+    {
+        $user = $this->getUser();
+        if (!$user instanceof User) {
+            throw $this->createAccessDeniedException('Invalid user');
+        }
+
+        $accesses = $user->getOpdsAccesses()->toArray();
+        if (count($accesses) === 0) {
+            throw $this->createAccessDeniedException('Invalid user, no opds access');
+        }
+        /** @var OpdsAccess $opdsAccess */
+        $opdsAccess = reset($accesses);
+
+        $params['accessKey'] = $params['accessKey'] ?? $opdsAccess->getToken();
+
+        return $this->generateUrl($route, $params, UrlGeneratorInterface::ABSOLUTE_URL);
+    }
+
+    #[Route('/', name: 'start')]
+    public function index(Request $request): Response
+    {
+        $opds = $this->opds->getOpdsConfig(''.$request->get('accessKey'));
+
+        $feeds = [];
+        $feeds[] = $this->opds->getNavigationEntry('books:series', 'Series', $this->generateOpdsUrl('opds_group', ['type' => 'serie']));
+        $feeds[] = $this->opds->getNavigationEntry('books:authors', 'Authors', $this->generateOpdsUrl('opds_group', ['type' => 'authors']));
+        $feeds[] = $this->opds->getNavigationEntry('books:tags', 'Tags', $this->generateOpdsUrl('opds_group', ['type' => 'tags']));
+        $feeds[] = $this->opds->getNavigationEntry('books:shelves', 'Shelves', $this->generateOpdsUrl('opds_shelves', []));
+        $feeds[] = $this->opds->getNavigationEntry('books:reading-list', 'My Reading List', $this->generateOpdsUrl('opds_readinglist', []));
+
+        $opds->feeds($feeds);
+
+        return $this->opds->convertOpdsResponse($opds->get()->getResponse());
+    }
+
+    #[Route('/search', name: 'search')]
+    public function search(Request $request): Response
+    {
+        $opds = $this->opds->getOpdsConfig(''.$request->get('accessKey'))->isSearch();
+        $opds->title('Search');
+
+        $books = $this->bookFinder->search(''.$request->get('q', $request->get('query', '')))->getResults();
+
+        $feeds = [];
+        foreach ($books as $result) {
+            $feeds[] = $this->opds->convertBookToOpdsEntry($result);
+        }
+
+        $opds->feeds($feeds);
+
+        return $this->opds->convertOpdsResponse($opds->get()->getResponse());
+    }
+
+    #[Route('/group/{type}', name: 'group')]
+    public function group(Request $request, string $type): Response
+    {
+        $group = match ($type) {
```

Comment:
Maybe add a todo to paginate ? 
It seems wrong to me to return everything.

I asked some AI and it seems you can define the `next` , `self`, `last` metadata in the response to paginate it.

I think I would not inject the BookRepository in the controller, but create an intermediate service, that will return a paginated information about the books.